core/src/main/java/com/datastax/oss/driver/internal/core/metrics/DropwizardSessionMetricUpdater.java

```diff
@@ -97,4 +119,18 @@ public String buildFullName(SessionMetric metric, String profileName) {
       return () -> 0;
     }
   }
+
+  @Nullable
+  private static Cache<?, ?> getPreparedStatementCache(InternalDriverContext context) {
+    // By default, both the sync processor and the async one are registered and they share the same
+    // cache. But with a custom processor registry, there could be only one of the two present.
```

Comment:
You mean that when the user will implement their own processor it can extend Async OR Sync?

Reply:
I wonder if there is a possiblity of some kind of race condition. I mean is the ordering of ifs matters? We are firstly checking `CqlPrepareAsyncProcessor` and then Sync version.

Reply:
I don't see any concurrency issue, this is called at initialization time only. And the two processors should share the same cache anyway, I've added a comment to the sync processor to clarify.